File: backend/app/api/health.py
from fastapi import APIRouter, Response, status

from backend.app.api.dependencies import HealthHandlerDependency, HealthProbeDependency
from backend.app.schemas.health import DependenciesResponse, LivenessResponse, ReadinessResponse

router = APIRouter(prefix="/health", tags=["health"])


@router.get("/live", response_model=LivenessResponse)
async def live() -> LivenessResponse:
    return LivenessResponse()


@router.get("/ready", response_model=ReadinessResponse)
async def ready(
    response: Response,
    probe: HealthProbeDependency,
    handler: HealthHandlerDependency,
) -> ReadinessResponse:
    result = handler.readiness(await probe.snapshot())
    if result.status != "ready":
        response.status_code = status.HTTP_503_SERVICE_UNAVAILABLE
    return result


@router.get("/dependencies", response_model=DependenciesResponse)
async def dependencies(
    probe: HealthProbeDependency,
    handler: HealthHandlerDependency,
) -> DependenciesResponse:
    return handler.dependencies(await probe.snapshot())

File: frontend/src/api/health.ts
import { apiRequest } from "./client";
import type {
  DependenciesResponse,
  LivenessResponse,
  ReadinessResponse,
} from "../types/api";

export type { DependenciesResponse, LivenessResponse, ReadinessResponse };

export function fetchLiveness(signal?: AbortSignal): Promise<LivenessResponse> {
  return apiRequest<LivenessResponse>("/health/live", { signal });
}

export function fetchReadiness(signal?: AbortSignal): Promise<ReadinessResponse> {
  return apiRequest<ReadinessResponse>("/health/ready", { signal });
}

export function fetchDependencies(
  signal?: AbortSignal,
): Promise<DependenciesResponse> {
  return apiRequest<DependenciesResponse>("/health/dependencies", { signal });
}
